User page

gil392 opened the pull request.

gil392 committed:
Add right-top icon of user, which is the actual user connected to the app.
Move user-icon to its own reusable component.
Remove convoluted MainPage, instead using UserPage directly.

gil392 committed:
Post uses the user's id and not the entire user. Helps to avoid multiple sources of truth and easy to work with all around.

gil392 committed:
When hovering on the icon, show small border to signify it's clickable.
Added menu to the userIcon. todo: add logic, and in the top-right corner it covers the icon itself (because of position: absolute).

gil392 committed:
Only show the user actions for the top-right corner icon (not for the icons in the posts).

When choosing an action, close the menu

gil392 committed:
Create UserData, which represents the data of the specific user. You can edit the name if you're the actual user.

Created a generic text field for better reusability.

gil392 committed:
Use the generic text field in all the forms

gil392 committed:
Refactor all icon buttons to use GenericIconButton component, more readable and easy to change.

ItayFinkelstein reviewed (commented):
notice npm run build fails. nees to fix it

ItayFinkelstein commented on src/CommentsPage.tsx:
```diff
@@ -91,7 +86,7 @@ export default function CommentsPage(props: CommentsPageProps) {
         {props.post.comments.map((comment) => {
           return (
-            <Card sx={{ minWidth: 500, maxWidth: 545 }}>
+            <Card sx={{ minWidth: 500, maxWidth: 545 }} key={comment.message}>
```
why is this the key?

gil392 replied:
A unique value, helps to differentiate between the elements. Without it an error message is thrown in the console. However, because the message may not be unique, I'll add now comment id and use it as key.

gil392 committed:
Fixed build issues due to unused imports.

Added CommentsPage.module.css for repeated css.

Have keys as id to make sure it's unique. The id will be given from the db in the future.

gil392 committed:
Part of previous commit, accidently wasn't pushed.

ItayFinkelstein reviewed (approved):
good job
